### ISSUE TITLE
Bump Kernel to at least 5.4.2

### ISSUE THREAD

**Issue by gbarton** [rancher/k3os#663]
**Version All**
k3os version v0.11.1

**Describe the bug**
Some golang code we have is running into a kernel bug that was patched mid 2020:
https://github.com/golang/go/wiki/LinuxKernelSignalVectorBug
Please update any packaged kernels to at least 5.4.2

**Expected behavior**
Code doesn't crash

**Actual behavior**
code crashes just like in the above link

**Additional context**
The ISO build has this bug, when we takeover an ubuntu image we do not see this issue.  It looks like 1.19 RC's have bumped the kernel to 5.4.0, is there any way to bump it a little more to 5.4.2?

**Comment by bhale** [comment]
Are you actually experiencing a bug? 

Ubuntu LTS continues to advance their stable kernels while keeping the version number at .0. If you look in their change logs or dmesg you will see the higher version is merged.

[    0.000000] Linux version 5.4.0-54-generic (root@ea247868655d) (gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)) #60 SMP Fri Nov 20 17:09:10 UTC 2020 (Ubuntu 5.4.0-54.60-generic 5.4.65)

**Comment by gbarton** [comment]
Yes, we have golang code that is hitting this bug consistently using the ISO version of K3os.  We backed the kernel down to 5.0.0-43 and the bug went away.

I am having a hard time to find a way to see when the patch made it into the ubuntu kernel, not having any luck figuring it out.  Is 1.19.x rc on the latest ubuntu LTS kernel?  If so we can try patching that kernel into the 0.11.1 version and seeing if the problem goes away.

**Comment by bhale** [comment]
k3os is on 5.4.0-54.60 from Focal which is bumped up to the kernel upstream version 5.4.65 as of 49.53. 
  * Focal update: v5.4.64 upstream stable release (LP: #1895880)

https://changelogs.ubuntu.com/changelogs/pool/main/l/linux/linux_5.4.0-65.73/changelog

You can see the change log here with details of upstream version bumps and ubuntu applied patches. It would be helpful if you could identify a version on an Ubuntu 20.04 system that fixes your issue or identify a specific commit. Otherwise we are kind of guessing.

**Comment by gbarton** [comment]
Its looking like Ubuntu LTS has not patched this.  Do you all run the LTS kernel directly or do you build your own with patches?  Am still trying to find out when/where it got patched in ubuntu.

**Comment by bhale** [comment]
k3os runs the exact same kernel as Ubuntu LTS. There are no k3os specific patches.

**Comment by gbarton** [comment]
I dont pretend to understand the linux kernel, thats voodoo territory to me, but a co-worker pointed to this:
https://github.com/rancher/k3os-kernel/tree/focal/lts/5.4.0/patches
So you build the kernel from source matching a particular LTS version? Is the patches the ones for that kernel version your making?

Am mostly curiosity at this point, I think I need to go figure out why LTS itself doesn't have this fix in it (or if it does, why its not working), then report back here what version we can update to.  I think its out of your hands if your running latest LTS already.

**Comment by cdeadspine** [comment]
I do not quite understand how to deal with k3os kernel version vs pod image kernel version discrepancies.

For instance, I suspect usbip has been broken in the kernel for a long time until someone fixed it in kernel:5.8.5+
(https://bugzilla.kernel.org/show_bug.cgi?id=208267)

I tried a bunch of different images such as 
ubuntu:xenial-20210114
debian:buster-20210208
and finally 
alpine:3.13
Alpine package info seems to hint that they have very recent builds of usbip 
(https://pkgs.alpinelinux.org/packages?name=usbip-utils&branch=v3.13)

This version of alpine says it is kernel 5.10 (https://alpinelinux.org/posts/Alpine-3.13.0-released.html)
but inside the pod it reports the k3os kernel version of course 
```
uname -a
Linux 5.4.0-54-generic #60 SMP Fri Nov 20 17:09:10 UTC 2020 x86_64 Linux
```

usbipd --version seems to say the same thing 2.0 everywhere, which is maybe a different problem in that the version isn't getting properly updated? i dont know

So say for instance I think something was fixed in the kernel source tree around 5.8.5, do I need to wait until k3os kernel is beyond this version before i can run newer builds of applications? 

Or is an alpine pod that supposedly has a community release of just a few months ago running recent builds?

**Comment by frezbo** [comment]
With the kernel 5.4 version that K3OS ships, it's not possible to boot the OS using grub on Raspberry Pi 4 ARM64. I got u-boot booting into grub and then grub starting the k3os kernel but it seems it can't find the rootfs on the partition named `K3OS_STATE`. If I swap out the kernel to use the Raspberry Pi kernel which is 5.10 I can boot into k3os
